### ISSUE TITLE
[fluent-react] DATETIME formatting always uses default

### ISSUE THREAD

**Issue by hkasemir** [projectfluent/fluent.js#90]
fluent version 0.4.1
```
// ftl file
today-is = today is { DATETIME($date, month: "long", day: "numeric") }
```

```
// React component
...
  <Localized id='today-is' $date={new Date()}>
    <p>today is {(new Intl.DateTimeFormat("fr", {month: "long", day: "numeric"})).format()}</p>
  </Localized>
  <p>today is {(new Intl.DateTimeFormat("fr", {month: "long", day: "numeric"})).format()}</p>
...
```

result
<img width="163" alt="image" src="https://user-images.githubusercontent.com/10947914/32575286-6c811a58-c488-11e7-83bf-afdaac6dff1b.png">

node version 6.11.2

**Comment by zbraniecki** [comment]
Seems like an issue with fluent-react, since I can't reproduce it in fluent-web.

@stasm ?

**Comment by stasm** [comment]
I can't reproduce this. In https://github.com/projectfluent/fluent.js/commit/cce4d42997e33cb8d0960bd62a56f5db36ca190a I added a `DATETIME($date, month: "long", day: "numeric")` placeable to `fluent-react`'s `hello-world` example. I can see the full name of the month spelled out correctly.

**Comment by hkasemir** [comment]
I can see that this is not reproduced by the examples, however, I was able to create a very basic test repo that repro's all the issues: https://github.com/projectfluent/fluent.js/issues/91, https://github.com/projectfluent/fluent.js/issues/90, and https://github.com/projectfluent/fluent.js/issues/89

https://github.com/hkasemir/test-fluent
it's using node 8.9.1, and requires `npm link` to the local fluent-react

**Comment by hkasemir** [comment]
however I did just discover that if I also `npm link fluent`, issue #90 is solved.

**Comment by zbraniecki** [comment]
I was able to reproduce all three issues using Heidi's minimized testcase. In my case, `npm link fluent` did not fix issue #90.

**Comment by stasm** [comment]
Thanks for putting together the testcase, @hkasemir. I can confirm that `npm link fluent` fixes #90. This is thanks to #81 which landed on master but hasn't made it into a released version yet.

Linking both `fluent` and `fluent-react` also fixes #89 for me. I suspect this is because when they're both linked, Webpack bundles them together and the `instanceof FluentType` actually works.

**Comment by stasm** [comment]
Once #98 lands I'll publish a new version of `fluent` which will make it easier to set up `fluent-react` for testing.

**Comment by stasm** [comment]
I just published `fluent` 0.4.2 which fixed this.
